feat(endpoint): Pass input data with command line

marktwtn opened the pull request:
The arguments of the endpoint core function
send_transaction_information() can be passed with the command line.

Close #671.

splasky commented on endpoint/endpointComp/endpoint.c:
```diff
@@ -47,21 +47,67 @@ static void print_help(void) {
+  le_arg_SetStringVar(&address, NULL, "addr");
+  le_arg_SetStringVar(&next_address, NULL, "next-addr");
+  le_arg_SetStringVar(&private_key_str, NULL, "pvt-key");
+  le_arg_SetStringVar(&device_id, NULL, "dev-id");
```
The device id will be capture from device implementation. We don't need to set the device id from command line.

marktwtn replied:
Removed.

splasky commented on endpoint/endpointComp/endpoint.c:
```diff
@@ -47,21 +47,66 @@ static void print_help(void) {
+  const char* iv_str = NULL;
+  uint8_t iv[AES_IV_SIZE] = {0};
+
+  le_arg_SetIntVar(&value, NULL, "value");
```
Update the print_help function to provide informative message.

marktwtn replied:
Done.

splasky commented on endpoint/endpointComp/endpoint.c:
```diff
@@ -47,21 +47,66 @@ static void print_help(void) {
+  le_arg_SetStringVar(&tag, NULL, "tag");
+  le_arg_SetStringVar(&address, NULL, "addr");
+  le_arg_SetStringVar(&next_address, NULL, "next-addr");
+  le_arg_SetStringVar(&private_key_str, NULL, "pvt-key");
```
The private key would be capture from the device implementation.

marktwtn replied:
It has been removed.
It will use the default test value now.

splasky commented on endpoint/endpointComp/endpoint.c:
```diff
@@ -47,21 +47,66 @@ static void print_help(void) {
+  le_arg_SetStringVar(&address, NULL, "addr");
+  le_arg_SetStringVar(&next_address, NULL, "next-addr");
+  le_arg_SetStringVar(&private_key_str, NULL, "pvt-key");
+  le_arg_SetStringVar(&iv_str, NULL, "iv");
```
I think initialize vector should be calculated be endpoint itself, @HowJMay what do you think?

HowJMay replied:
It should be a random byte list. I think it can be generated every time endpoint sends a message

marktwtn replied:
The implementation of reading initialization vector from command line is removed.

splasky commented on endpoint/endpointComp/endpoint.c:
```diff
@@ -47,21 +47,66 @@ static void print_help(void) {
+  if (private_key_str)
+    string_to_uint8_arr(private_key_str, private_key);
+  else
+    memcpy(private_key, test_private_key, 32);
```
Change the constant value to AES_CBC_KEY_SIZE.

marktwtn replied:
Done.

splasky commented on endpoint/endpointComp/endpoint.c:
```diff
@@ -47,21 +47,66 @@ static void print_help(void) {
+  const char* address = TEST_ADDRESS;
+  const char* next_address = TEST_NEXT_ADDRESS;
+  const char* private_key_str = NULL;
+  uint8_t private_key[32] = {0};
```
Change the constant value to AES_CBC_KEY_SIZE.

marktwtn replied:
Done.

splasky commented on endpoint/endpointComp/endpoint.c:
```diff
@@ -25,8 +25,8 @@
 
-const uint8_t test_key[32] = {82,  142, 184, 64,  74, 105, 126, 65,  154, 116, 14,  193, 208, 41,  8,  115,
-                              158, 252, 228, 160, 79, 5,   167, 185, 13,  159, 135, 113, 49,  209, 58, 68};
+const uint8_t test_private_key[32] = {82,  142, 184, 64,  74, 105, 126, 65,  154, 116, 14,  193, 208, 41,  8,  115,
```
Change the constant value to AES_CBC_KEY_SIZE.

marktwtn replied:
Done.

splasky commented:
Change the title from test to feat.

jserv commented on endpoint/endpointComp/endpoint.c:
```diff
@@ -37,31 +38,98 @@ static void print_help(void) {
+    str_start += (str_bytes + 1);
+    arr_idx++;
+  }
+  return;
```
The trailing `return` is redundant.

marktwtn replied:
Removed.

HowJMay commented on endpoint/endpointComp/endpoint.c:
```diff
@@ -37,31 +38,97 @@ static void print_help(void) {
+      "           [--msg-fmt=\"message-format\"]\n"
+      "           [--addr=\"address\"]\n"
+      "           [--next-addr=\"next-address\"]\n"
+      "           [--iv=\"initialize-vector\"]\n"
```
Initialization vector

marktwtn replied:
Removed.

HowJMay commented on endpoint/endpointComp/endpoint.c:
```diff
@@ -37,31 +38,97 @@ static void print_help(void) {
+      "    The \"next-address\" should be composed with [9A-Z] and the length should be 81.\n"
+      "\n"
+      "  --iv=\"initialize-vector\"\n"
+      "    Assign initialize vector with string for send_transaction_information.\n"
```
ditto

marktwtn replied:
Removed.

HowJMay commented on endpoint/endpointComp/endpoint.c:
```diff
@@ -37,31 +38,97 @@ static void print_help(void) {
+      "\n"
+      "  --iv=\"initialize-vector\"\n"
+      "    Assign initialize vector with string for send_transaction_information.\n"
+      "    The \"initialize-vector\" format is \"number,number,...,number\".\n"
```
ditto

marktwtn replied:
Removed.

HowJMay commented on endpoint/endpointComp/endpoint.h:
```diff
@@ -9,9 +9,19 @@
+ * @brief Convert the string to uint8_t array
+ *
+ * @param[in] string The string with the combination of number and comma
+ * @param[in,out] arr The array for storing the converted number
```
converted numbers

marktwtn replied:
Removed.

HowJMay commented on endpoint/endpointComp/endpoint.h:
```diff
@@ -9,9 +9,19 @@
+/**
+ * @brief Convert the string to uint8_t array
+ *
+ * @param[in] string The string with the combination of number and comma
```
numbers and commas

marktwtn replied:
Removed.

marktwtn commented:
I added the missing code related to `tag`.
I also removed the code related to `initialization vector`, which would be generated with random byte in the future instead of reading from command line input.
The function `string_to_uint8_arr()` is also removed since it is no longer used.

marktwtn commented:
If there is no problem, the pull request can be merged.
